api/src/test/java/org/openmrs/api/CohortServiceTest.java

```diff
@@ -521,8 +526,128 @@ public void removePatientFromCohort_shouldSaveCohortAfterRemovingPatient() throw
-		assertFalse(service.getCohort(2).contains(patientToAddThenRemove));
+		List<CohortMembership> memberList = service.getCohort(2)
+				.getMembers().stream()
+				.filter(m -> m.getPatient().getPatientId().equals(patientToAddThenRemove.getPatientId()))
+				.collect(Collectors.toList());
+		CohortMembership memberWithPatientToRemove = memberList.get(0);
+		assertNotNull(memberWithPatientToRemove.getEndDate());
+	}
+
+	/**
+	 * @verifies {@link CohortService#addMembershipToCohort(Cohort, CohortMembership)}
+	 * test = should add membership to cohort
+	 */
+	@Test
+	@Verifies(value = "should add membership to cohort", method = "addMembershipToCohort(Cohort, CohortMembership)")
+	public void addMembershipToCohort_shouldAddMembershipToCohort() throws Exception {
+		executeDataSet(COHORT_XML);
+		
+		Patient p = new Patient(4);
+		if (!service.getCohort(1).contains(p)) {
+			CohortMembership memberToAdd = new CohortMembership(p);
+			service.addMembershipToCohort(service.getCohort(1), memberToAdd);
+			assertTrue(service.getCohort(1).contains(p));
+		}
+	}
+
+	/**
+	 * @verifies {@link CohortService#removeMemberShipFromCohort(Cohort, CohortMembership)}
+	 * test = should remove membership from cohort
+	 */
+	@Test
+	@Verifies(value = "should remove membership from cohort", method = "removeMemberShipFromCohort(Cohort, CohortMembership)")
+	public void removeMembershipFromCohort_shouldRemoveMembershipFromCohort() throws Exception {
+		executeDataSet(COHORT_XML);
+
+		CohortMembership memberToAddThenRemove = new CohortMembership(new Patient(4));
+		service.addMembershipToCohort(service.getCohort(1), memberToAddThenRemove);
+		assertTrue(service.getCohort(1).contains(memberToAddThenRemove.getPatient()));
+		assertNull(memberToAddThenRemove.getEndDate());
+
+		service.removeMemberShipFromCohort(service.getCohort(1), memberToAddThenRemove);
+		assertNotNull(memberToAddThenRemove.getEndDate());
+	}
+
+	/**
+	 * @verifies {@link CohortService#patientVoided(Patient)}
+	 * test = should void membership for every patient voided
+	 */
+	@Test
+	@Verifies(value = "should void membership for every patient voided", method = "patientVoided(Patient)")
+	public void patientVoided_shouldVoidMemberships() throws Exception {
+		executeDataSet(COHORT_XML);
+
+		Cohort cohort = Context.getCohortService().getCohort(2);
+		Patient voidedPatient = new Patient(7);
+		voidedPatient.setVoided(true);
+		voidedPatient.setDateVoided(new Date());
+		voidedPatient.setVoidedBy(Context.getAuthenticatedUser());
+		voidedPatient.setVoidReason("Voided as a result of the associated patient getting voided");
+
+		CohortMembership newMemberContainingVoidedPatient = new CohortMembership(voidedPatient);
+		cohort.addMembership(newMemberContainingVoidedPatient);
+		assertTrue(cohort.contains(voidedPatient));
+
+		service.patientVoided(voidedPatient);
+		assertEquals(voidedPatient.getVoided(), newMemberContainingVoidedPatient.getVoided());
```

Comment:
Why not use assertTrue? This will make it explicitly clear that you expect the patient to be voided.